src/GraphQL/GraphQLExtensions.cs

```diff
@@ -281,7 +281,23 @@ public static string[] IsValidLiteralValue(this IGraphType type, IValue valueAst
 
             var scalar = (ScalarGraphType)type;
 
-            var parseResult = scalar.ParseLiteral(valueAst);
+            object parseResult = null;
+            try
+            {
+                if (scalar is ICanParseScalar parseable)
+                {
+                    if (parseable.CanParseLiteral(valueAst)) // it still can throw
+                        return Array.Empty<string>();
```

Comment:
Returns empty error array in a good case, otherwise `parseResult` just remains `null` without generating any exception as it was before.